work5/file_1.py

```diff
@@ -0,0 +1,62 @@
+"""
+Необходимо создать API для управления списком задач.
+Каждая задача должна содержать заголовок и описание.
+Для каждой задачи должна быть возможность указать статус (выполнена/не выполнена).
+
+API должен содержать следующие конечные точки:
+— GET /tasks — возвращает список всех задач.
+— GET /tasks/{id} — возвращает задачу с указанным идентификатором.
+— POST /tasks — добавляет новую задачу.
+— PUT /tasks/{id} — обновляет задачу с указанным идентификатором.
+— DELETE /tasks/{id} — удаляет задачу с указанным идентификатором.
+
+Для каждой конечной точки необходимо проводить валидацию данных запроса и ответа.
+Для этого использовать библиотеку Pydantic.
+"""
+from fastapi import FastAPI, Request
+from fastapi.responses import HTMLResponse
+from fastapi.templating import Jinja2Templates
+import pandas as pd
+from hw.work5.model import Tasks
+
+app = FastAPI()
+templates = Jinja2Templates(directory='hw/work5/templates')
+list_ = []
+
+
+@app.get('/tasks/', response_class=HTMLResponse)
+def list_of_tasks(requ: Request):
+	"""Выкладываем список задач"""
+	table = pd.DataFrame([vars(task) for task in list_]).to_html()
+	return templates.TemplateResponse('tasks_html.html', {'request': requ, 'table': table})
+
+
+@app.get('/tasks/{num}', response_model=Tasks)
+def select_task(num: int):
+	"""Выбор задачи по id"""
+	task_num = [task for task in list_ if task.id == num]
+	return task_num[0]
+
+
+@app.post('/tasks/', response_model=Tasks)
+def set_task(task: Tasks):
+	"""Добавляем задачи"""
+	task.id = len(list_) + 1
+	list_.append(task)
+	return task
+
+
+@app.put('/tasks/{id}', response_model=Tasks)
+def update_task(id: int, task: Tasks):
+	"""Обновляем задачи"""
+	task.id = id
+	list_[id - 1] = task # если введут id, превышающий количество задач - выпадет ошибка
+	return task
```

Comment:
Все верно, потому что index и уникальный идентификатор – не одно и то же
```suggestion
        for i, task_db in list_:
            if task_db.id == id:
	    task.id = id
	    list_[i] = task
	    return task
```
А еще у вас отступы были неверные. Поправил

Reply:
1) честно говоря я так и не понял для чего нужен здесь цикл.
Нам не надо искать совпадение индексов - мы просто подменяем имеющуюся задачу на прилетевшую задачу с  тем же индексом!
2) так и не увидел где у меня отступы неверные.А вот у вас Данил наоборот в предложенном цикле неверный отступ в условии в 54,55,56 строках

Reply:
Цикл нужен для того, чтобы правильно обновлять задачу. Если вы вначале создали несколько задач, то их id будет на единицу больше, чем их индекс в списке. Но если после этого вы удалите несколько задач из начала списка, то индекс остальных задач пересчитается, а id останутся такими же. Поэтому правильный способ поиска задачи это искать ее по id из списка, а для этого нужно перевидать все задачи в цикле.

Тут, кстати, еще одна проблема возникает. Если мы удалили несколько старых задач, а после попытались добавить новые, то в список добавятся задачи с id, который уже есть в этом списке. Это все из-за способа задания id в 44 строке. Но это можно списать на совершенство условия задания, которое мы исправили на последнем семинаре.

По какой-то причине у вас отступ был больше в 2 раза от того, каким он должен быть. Один уровень отступа в Python – 4 пробельных символа. Но у вас в коде пробелов вообще не было. Только ощущение, как будто отступы создавались при помощи клавиши tab. Но вы правильно заметили, что я сам забыл поставить свои отступы правильно))) Исправил сообщение, теперь они нормальные, такие как должны быть.